sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1661,6 +1663,83 @@ def get_custom_materialization_type(name: str) -> t.Type[CustomMaterialization]:
     return strategy_type
 
 
+class EngineManagedStrategy(PromotableStrategy):
+    def create(
+        self,
+        table_name: str,
+        model: Model,
+        is_table_deployable: bool,
+        **render_kwargs: t.Any,
+    ) -> None:
+        rendered_query = model.render_query_or_raise(**render_kwargs)
+
+        # only create the managed table if this snapshot can actually be deployed to prod (eg if an upstream model doesnt have a forward-only change)
+        # vendors providing managed tables (such as Snowflake's dynamic tables) tend to charge more for them, they arent as cheap as normal tables
+        if is_table_deployable:
+            logger.info("Creating managed table: %s", table_name)
+            self.adapter.create_engine_managed_table(
```

Comment:
Shouldn't we defer the creation of the deployable table until the change is promoted to production?